pkg/ruler/api_test.go

```diff
@@ -27,125 +27,219 @@ import (
 )
 
 func TestRuler(t *testing.T) {
+	const (
+		userID   = "user1"
+		interval = time.Minute
+	)
+
 	testCases := map[string]struct {
-		mockRules        map[string]rulespb.RuleGroupList
-		userID           string
-		expectedResponse response
+		configuredRules rulespb.RuleGroupList
+		overrides       map[string]rulerLimits
+		expectedRules   []*RuleGroup
 	}{
-		"rules": {
-			mockRules: mockRules,
-			userID:    "user1",
-			expectedResponse: response{
-				Status: "success",
-				Data: &RuleDiscovery{
-					RuleGroups: []*RuleGroup{
-						{
-							Name: "group1",
-							File: "namespace1",
-							Rules: []rule{
-								&recordingRule{
-									Name:   "UP_RULE",
-									Query:  "up",
-									Health: "unknown",
-									Type:   "recording",
-								},
-								&alertingRule{
-									Name:   "UP_ALERT",
-									Query:  "up < 1",
-									State:  "inactive",
-									Health: "unknown",
-									Type:   "alerting",
-									Alerts: []*Alert{},
-								},
-							},
-							Interval: 60,
+		"should load and evaluate the configured rules": {
+			configuredRules: rulespb.RuleGroupList{
+				&rulespb.RuleGroupDesc{
+					Name:      "group1",
+					Namespace: "namespace1",
+					User:      "user1",
+					Rules:     []*rulespb.RuleDesc{mockRecordingRuleDesc("UP_RULE", "up"), mockAlertingRuleDesc("UP_ALERT", "up < 1")},
+					Interval:  interval,
+				},
+			},
+			expectedRules: []*RuleGroup{
+				{
+					Name: "group1",
+					File: "namespace1",
+					Rules: []rule{
+						&recordingRule{
+							Name:   "UP_RULE",
+							Query:  "up",
+							Health: "unknown",
+							Type:   "recording",
+						},
+						&alertingRule{
+							Name:   "UP_ALERT",
+							Query:  "up < 1",
+							State:  "inactive",
+							Health: "unknown",
+							Type:   "alerting",
+							Alerts: []*Alert{},
 						},
 					},
+					Interval: 60,
 				},
 			},
 		},
-		"rules special characters": {
-			mockRules: mockSpecialCharRules,
-			userID:    "user1",
-			expectedResponse: response{
-				Status: "success",
-				Data: &RuleDiscovery{
-					RuleGroups: []*RuleGroup{
-						{
-							Name: ")(_+?/|group1+/?",
-							File: ")(_+?/|namespace1+/?",
-							Rules: []rule{
-								&recordingRule{
-									Name:   "UP_RULE",
-									Query:  "up",
-									Health: "unknown",
-									Type:   "recording",
-								},
-								&alertingRule{
-									Name:   "UP_ALERT",
-									Query:  "up < 1",
-									State:  "inactive",
-									Health: "unknown",
-									Type:   "alerting",
-									Alerts: []*Alert{},
-								},
-							},
-							Interval: 60,
+		"should load and evaluate only recording rules if alerting rules evaluation is disabled for the tenant": {
+			configuredRules: rulespb.RuleGroupList{
+				&rulespb.RuleGroupDesc{
+					Name:      "group1",
+					Namespace: "namespace1",
+					User:      "user1",
```

Comment:
```suggestion
					User:      userID,
```

Reply:
aside: what will happen if I forget to add a user to my test case? Should we verify that  `RuleGroupDesc.User == userID`  for each test case? Not sure how much of a safety net we need for contributors.

Reply:
Currently there's no significant impact (it's used in few logs, and in part of the code not triggered by these tests). However, who know how code will change in the future? :) I will add a pre-condition check as you suggested.